Source/Core/Core/MemoryWatcher.cpp

```diff
@@ -57,24 +70,44 @@ bool MemoryWatcher::OpenSocket(const std::string& path)
 	return m_fd >= 0;
 }
 
+u32 MemoryWatcher::ChasePointer(const std::string& line)
+{
+	u32 value = 0;
+	for (u32 offset : m_addresses[line])
+		value = Memory::Read_U32(value + offset);
+	return value;
+}
+
 void MemoryWatcher::WatcherThread()
 {
 	while (m_running)
 	{
 		for (auto& entry : m_values)
 		{
-			u32 address = entry.first;
+			std::string address = entry.first;
 			u32& current_value = entry.second;
 
-			u32 new_value = Memory::Read_U32(address);
+			u32 new_value = ChasePointer(address);
 			if (new_value != current_value)
 			{
+				// Update the value
 				current_value = new_value;
-				u32 buf[2] = {address, current_value};
+
+				// Compose the message
+				std::stringstream message_stream;
+				message_stream << address << '\n' << std::hex << current_value;
+				std::string message = message_stream.str();
+
+				// Pack it into a buffer and send it. Don't allow overflow.
+				char buf[BUFFER_SIZE];
+				if (message.size() + 1 > BUFFER_SIZE)
+					continue;
+				std::size_t offset = message.copy(buf, message.size());
```

Comment:
Why is there even a copy of the whole string being made? Just use substr and pass c_str() of the resulting string to the send to call (you will need to change the static_cast to `const void*`